pytorch_toolkit/ote/ote/modules/arg_converters/base.py

```diff
@@ -15,97 +15,184 @@
-        'gpu_num': 'gpu_num',
-        'tensorboard_dir': 'tensorboard_dir',
-        'nncf_quantization': 'nncf_quantization',
-        'nncf_pruning': 'nncf_pruning',
-        'nncf_sparsity': 'nncf_sparsity',
-        'nncf_binarization': 'nncf_binarization',
-    }
-    test_out_args_map = {
-        'load_weights': 'snapshot',
-        'save_metrics_to': 'out',
-        'save_output_to': 'show_dir',
-    }
-
-    def __init__(self):
+from abc import ABCMeta, abstractmethod
+from collections import namedtuple
+
+
+def map_args(src_args, args_map):
+    return {v: src_args[k] for k, v in args_map.items()}
+
+class ArgConverterMaps(metaclass=ABCMeta):
+    @abstractmethod
+    def train_update_args_map(self):
+        """
+        Returns a map:
+            from: the name of cmd line arguments of the corresponding **train** argparser.parse_args
+                  (e.g. `argparser = DefaultArgParser.get_train_parser()`)
+            to:   the fields of mmdetection/mmaction/other config for its **train** script in the form
+                  compatible with mmcv.Config.merge_from_dict
+        """
         pass
 
-    def convert_train_args(self, model_template_path, args):
-        update_args = self.__map_args(args, self.train_update_args_map)
+    @abstractmethod
+    def test_update_args_map(self):
+        """
+        Returns a map:
+            from: the name of cmd line arguments of the corresponding **test** argparser.parse_args
+                  (e.g. `argparser = DefaultArgParser.get_test_parser()`)
+            to:   the fields of mmdetection/mmaction/other config for its **test** script in the form
+                  compatible with mmcv.Config.merge_from_dict
+        """
+        pass
 
-        extra_args = self._get_extra_train_args(args)
-        update_args.update(extra_args)
+    @abstractmethod
+    def compress_update_args_map(self):
+        """
+        Returns a map:
+            from: the name of cmd line arguments of the corresponding **compress** argparser.parse_args
+                  (e.g. `argparser = DefaultArgParser.get_compression_parser()`)
+            to:   the fields of mmdetection/mmaction/other config for its **train** script in the form
+                  compatible with mmcv.Config.merge_from_dict to makes compression
+                  (note that compression makes the original train script with special tuned config;
+                   typically the parameters are the same as for training except
+                   learning rate and total_epochs parameters)
+
+        """
+        pass
 
-        template_folder = os.path.dirname(model_template_path)
-        converted_args = {
-            'config': os.path.join(template_folder, args['config']),
-            'out': os.path.join(args['save_checkpoints_to'], model_template_path),
-            'update_config': update_args,
-        }
-        converted_args.update(self.__map_args(args, self.train_out_args_map))
+    @abstractmethod
```

Comment:
Is it really necessary to declare `abstractmethod` here if most of `arg_converters` will use "as is"? We can reduce copy-paste by removing it.

Reply:
I like to be sure that nobody forget to implement/override something.
So, I force the person who makes a derived class to implement all of the required methods.

IMHO, the copy of code is small -- just one line per function.
But if Ilya also supposes that it is not required -- OK, I'll remove `@abstractmethod` here

Reply:
2 @Ilya-Krylov : Ilya, what is your opinion?

Reply:
Anyway if somebody know what is he doing he also know (can find) where to change it. The rest people don't need to re-implement each line of base class...

Reply:
@LeonidBeynenson I agree with @evgeny-izutov

Reply:
Ok!

Reply:
Fixed